tasks/main.yml

```diff
@@ -96,18 +116,29 @@
     force_basic_auth: true
   register: grafana_conf_existing_dashboards
 
-- name: Find available dashboards
+- name: Find dashboards from specified git repo
+  find:
+    path: '{{ grafana_conf_grafana_dashboard_repo.checkout_path ~ grafana_conf_grafana_dashboard_repo.relative_path }}'
+    file_type: file
+    patterns: '*.json'
+  register: grafana_conf_available_dashboards_git
+  when: grafana_conf_grafana_dashboard_repo.checkout_path is defined
+
+- name: Find dashboards from specified path
   find:
-    path: '{{ grafana_conf_grafana_dashboards }}'
+    path: '{{ grafana_conf_grafana_dashboard_path }}'
     file_type: file
     patterns: '*.json'
-  register: grafana_conf_available_dashboards
+  register: grafana_conf_available_dashboards_path
+  when: grafana_conf_grafana_dashboard_path != ""
 
 - name: Slurp available dashboards
   slurp:
     src: "{{ item['path'] }}"
   register: grafana_conf_available_dashboard_contents
-  with_items: "{{ grafana_conf_available_dashboards['files'] }}"
+  with_items:
+    - "{{ grafana_conf_available_dashboards_git['files'] | default([]) }}"
+    - "{{ grafana_conf_available_dashboards_path['files'] | default([]) }}"
```

Comment:
This is now a nested list, does that still work, or should we be using with_items over a concatenated list, or using with_nested?

Reply:
I believe it works although it would be clearer to explicitly concatenate them. Much to some peoples dislike it flattens the list of lists:  https://github.com/ansible/ansible/issues/5913